crates/vision/src/image_segmenter.rs

```diff
@@ -109,20 +125,65 @@ impl ImageSegmenter {
     }
 }
 
+fn padding_size(median_mode: MedianModeParameters) -> u32 {
+    match median_mode {
+        MedianModeParameters::Disabled => 0,
+        MedianModeParameters::ThreePixels => 1,
+        MedianModeParameters::FivePixels => 2,
+    }
+}
+
 #[allow(clippy::too_many_arguments)]
 fn new_grid(
     image: &YCbCr422Image,
+    camera_matrix: Option<&CameraMatrix>,
     horizon: &Horizon,
     field_color: &FieldColorParameters,
     horizontal_stride: usize,
+    horizontal_edge_threshold: i16,
+    horizontal_median_mode: MedianModeParameters,
+    horizontal_edge_detection_source: EdgeDetectionSourceParameters,
     vertical_stride: usize,
+    vertical_stride_in_robot_coordinates: f32,
     vertical_edge_detection_source: EdgeDetectionSourceParameters,
     vertical_edge_threshold: i16,
     vertical_median_mode: MedianModeParameters,
     projected_limbs: &[Limb],
 ) -> ScanGrid {
+    let horizontal_padding_size = padding_size(horizontal_median_mode);
+    let vertical_padding_size = padding_size(vertical_median_mode);
+
+    let horizon_y_maximum = horizon
+        .horizon_y_maximum()
+        .clamp(0.0, image.height() as f32);
```

Comment:
Does this need to be clamped? 🤔

Reply:
the horizon can be well outside of the image. I think it doesn't make sense to segment there.